routes/notificationRouter.js

```diff
@@ -1,8 +1,83 @@
-// const express = require('express');
-// const { db } = require('../server/db');
+const express = require('express');
+const { db } = require('../server/db');
 
-// const notificationRouter = express.Router();
+const notificationRouter = express.Router();
 
-// Your code below (delete this line when committing):
+module.exports = notificationRouter;
 
-// module.exports = notificationRouter;
+notificationRouter.get('/', async (req, res) => {
+  try {
+    const allNotifications = await db.query('SELECT * FROM notification;');
+    res.status(200).send(allNotifications);
+  } catch (err) {
+    res.status(500).json({
+      status: 'Failed',
+      msg: err.message,
+    });
+  }
+});
+
+notificationRouter.get('/:id', async (req, res) => {
+  const { id } = req.params;
+  try {
+    const idNotification = await db.query('SELECT * FROM notification WHERE business_id = $(id);', {
+      id,
+    });
+    res.status(200).send(idNotification);
+  } catch (err) {
+    res.status(500).json({
+      status: 'Failed',
+      msg: err.message,
+    });
+  }
+});
+
+notificationRouter.post('/', async (req, res) => {
+  const { business_id: businessId, message, timestamp, been_dismissed: beenDismissed } = req.body;
+  try {
+    await db.query(
+      `
+        INSERT INTO notification (business_id, message, timestamp, been_dismissed)
+        VALUES
+        ($(businessId), $(message), $(timestamp), $(beenDismissed));
+      `,
+      { businessId, message, timestamp, beenDismissed },
+    );
+    res.status(200).json({
+      status: 'Success',
+    });
+  } catch (err) {
+    res.status(500).json({
+      status: 'Failed',
+      msg: err.message,
+    });
+  }
+});
+
+notificationRouter.put('/:id', async (req, res) => {
+  const { id } = req.params;
+
+  const { business_id: businessId, message, timestamp, been_dismissed: beenDismissed } = req.body;
+
+  const updateNotification = await db.query(
+    `UPDATE notification SET
+      business_id = $(businessId),
+      message = $(message),
```

Comment:
In a put request, the values being changed should be optional because not all values are guaranteed to be updated each request.

Here is an example on how you can make each value optional.

```
const updatedDonation = await db.query(
      `UPDATE donation_tracking SET
      donation_id = $(donationID)
      ${businessID ? `, business_id = $(businessID) ` : ''}
      ${foodBankDonation ? `, food_bank_donation = $(foodBankDonation) ` : ''}
      ${reporter ? `, reporter = $(reporter) ` : ''}
      ${email ? `, email = $(email) ` : ''}
      WHERE donation_id = $(donationID)
      RETURNING *;`,
      { businessID, donationID, foodBankDonation, reporter, email },
    );
    ```